src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/DateTrunc.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MartinGeorgiev\Doctrine\ORM\Query\AST\Functions;
+
+use Doctrine\ORM\Query\AST\Literal;
+use Doctrine\ORM\Query\AST\Node;
+use MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\Exception\InvalidTruncFieldException;
+use MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\Traits\TimezoneValidationTrait;
+
+/**
+ * Implementation of PostgreSQL DATE_TRUNC().
+ *
+ * @see https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC
+ * @since 3.7
+ *
+ * @author Jan Klan <jan@klan.com.au>
+ *
+ * @example Using it in DQL: "SELECT DATE_TRUNC('day', e.timestampWithTz, 'Australia/Adelaide') FROM Entity e"
+ */
+class DateTrunc extends BaseVariadicFunction
+{
+    use TimezoneValidationTrait;
+    /**
+     * @see https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC
+     */
+    public const FIELDS = [
```

Comment:
Can you elaborate on why this is a public constant? There is no such pattern in any other classes in the repository. On the surface, it is needed for tests primarily.

Reply:
I wanted to both (a) enumerate the allowed options in [InvalidTruncFieldException message](https://github.com/martin-georgiev/postgresql-for-doctrine/pull/493/files#diff-c412d780a518fb4dad7058aaf6cf274ba0c7828f4d50049c7282da681ad26ee5) and (b) have the list of valid options available for input validation. 

It's in a public constant because it will likely never change (hence not a static variable), it's exposed as a class constant because it's needed in two separate classes, and it's not hardcoded as two separate arrays in each class because if it does change, someone will have to remember changing it in both places. 

At the end of the day if you think it's better as two separate hardcoded arrays (because it's unlikely to change), let's do it that way for sake of overall consistency.

Reply:
I prefer PostgreSQL to throw its native error message and wrap it, rather than trying to validate lower-level PostgreSQL behaviour.